crates/swc_ecma_minifier/src/compress/optimize/iife.rs

```diff
@@ -200,6 +200,7 @@ where
 
                         if let Some(arg) = arg {
                             match &**arg {
+                                Expr::Lit(Lit::Regex(..)) => continue,
```

Comment:
This likely means that any non-literal expression is unsafe. Eg,

```js
const r = new Regex('a', 'g');
function run(str, r) {
    let m
    while (m = r.exec(str)) {
        console.log(m)
    }
}
run('abcda', r)

Reply:
`_ => continue,` below will catch it and do same thing (`continue`

Reply:
I'll add a test anyway

Reply:
Perfect, I didn't notice it.

Reply:
I added one